options/options.go

```diff
@@ -0,0 +1,23 @@
+package options
+
+type validationFunc func(val interface{}) bool
+
+type optionDetails struct {
+	defaultValue    interface{}
+	validationFuncs []validationFunc
+	help            string
+}
+
```

Comment:
We intend to have xlators export their option table in a consumable manner. I'd rather have us not hardcode any sort of option table in GD2 from the beginning itself. But I'm okay with this for now.